File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, l: List[int]) -> int:
        sum=l[0]
        ans=sum
        for i in range(1,len(l)):
            if(sum<l[i] and sum<0):
                sum=0
            sum+=l[i]
            if(sum>ans):
                ans=sum
        return(ans)
        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& v) 
    {
        int sum=v[0];
        int ans=sum;
        int i;
        for(int i=1;i<v.size();i++)
        {
            if(v[i]>sum && sum<0)
            {
                sum=0;
            }
            sum+=v[i];
            if(sum>ans)
            {
                ans=sum;
            }
            
        }
        return(ans);
        
    }
};